### PR TITLE
Impelmented basic EoE 

### PR DESCRIPTION
**Pull request by nakarlsson**
Impelmented basic EoE 
- set IP
- get IP
- send Ethernet frame
- read Ethrnernet frame fragment via mailbox hook

### CODE REVIEW

**Comment by nakarlsson** [comment]
First version that make use of current mailbox functionality. mailbox receive hook added handles EoE frames that can arrive at any time for any EoE slave, to be handled by the user when it comes to saving fragment data, fragment number and frame no in application data.

File: soem/ethercateoe.h [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+int ecx_EOEreadfragment(
+   ec_mbxbuft * MbxIn,
+   uint8 * rxfragmentno,
+   uint16 * rxframesize,
+   uint16 * rxframeoffset,
+   uint16 * rxframeno,
+   int * psize,
+   void *p);
+
+#ifdef EC_VER1
```
Comment (ArthurKetels) [inline soem/ethercateoe.h]: I think it is better to just skip VER1 for new functions. This is new functionality and new users should always use the new interface.

File: soem/ethercatmain.c [hunk 2]
```diff
@@ -1052,6 +1051,25 @@ int ecx_mbxreceive(ecx_contextt *context, uint16 slave, ec_mbxbuft *mbx, int tim
                   wkc = 0; /* prevent emergency to cascade up, it is already handled. */
                }
             }
+            else if ((wkc > 0) && ((mbxh->mbxtype & 0x0f) == ECT_MBXT_EOE)) /* EoE response? */
+            {
+               ec_EOEt * eoembx = (ec_EOEt *)mbx;
+               uint16 frameinfo1 = etohs(eoembx->frameinfo1);
+               /* All non fragement data frame types are expected to be handled by
```
[the remark this answers is not in the document]
Comment (nakarlsson) [reply soem/ethercatmain.c]: Yes

File: test/linux/eoe_test/eoe_test.c [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+      /* create thread to handle slave error handling in OP */
+//      pthread_create( &thread1, NULL, (void *) &ecatcheck, (void*) &ctime);
+      osal_thread_create(&thread1, 128000, &ecatcheck, (void*) &ctime);
+
+      /* start cyclic part */
+      teststarter(argv[1]);
+   }
+   else
+   {
+      printf("Usage: simple_test ifname1\nifname = eth0 for example\n");
```
[the remark this answers is not in the document]
Comment (nakarlsson) [reply test/linux/eoe_test/eoe_test.c]: Fixed

File: soem/ethercatmain.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (nakarlsson) [reply soem/ethercatmain.c]: Fixed

File: test/linux/eoe_test/eoe_test.c [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+         printf("\nRequest init state for all slaves\n");
+         ec_slave[0].state = EC_STATE_INIT;
+         /* request INIT state for all slaves */
+         ec_writestate(0);
+      }
+      else
+      {
+         printf("No slaves found!\n");
+      }
+      printf("End simple test, close socket\n");
```
[the remark this answers is not in the document]
Comment (nakarlsson) [reply test/linux/eoe_test/eoe_test.c]: Fixed

File: test/linux/eoe_test/eoe_test.c [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+   osal_thread_create(&thread2, 128000, &mailbox_reader, &ecx_context);
+}
+
+void teststarter(char *ifname)
+{
+   int i, oloop, iloop, chk;
+   needlf = FALSE;
+   inOP = FALSE;
+
+   printf("Starting simple test\n");
```
[the remark this answers is not in the document]
Comment (nakarlsson) [reply test/linux/eoe_test/eoe_test.c]: Fixed

File: soem/ethercateoe.h (hunk 1 above)
Comment (nakarlsson) [reply soem/ethercateoe.h]: Fixed

**Pull request merged**